com.unity.multiplayer.mlapi/Runtime/Core/SnapshotSystem.cs

```diff
@@ -0,0 +1,412 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using MLAPI.Configuration;
+using MLAPI.Messaging;
+using MLAPI.NetworkVariable;
+using MLAPI.Serialization;
+using MLAPI.Serialization.Pooled;
+using MLAPI.Transports;
+using UnityEngine;
+using UnityEngine.UIElements;
+
+namespace MLAPI
+{
+    // Structure that acts as a key for a NetworkVariable
+    // Allows telling which variable we're talking about.
+    // Might include tick in a future milestone, to address past variable value
+    internal struct Key
+    {
+        public ulong NetworkObjectId; // the NetworkObjectId of the owning GameObject
+        public ushort BehaviourIndex; // the index of the behaviour in this GameObject
+        public ushort VariableIndex; // the index of the variable in this NetworkBehaviour
+    }
+
+    // Index for a NetworkVariable in our table of variables
+    // Store when a variable was written and where the variable is serialized
+    internal struct Entry
+    {
+        public Key Key;
+        public ushort TickWritten; // the network tick at which this variable was set
+        public ushort Position; // the offset in our Buffer
+        public ushort Length; // the length of the data in Buffer
+        public bool Fresh; // indicates entries that were just received
+
+        public const int NotFound = -1;
+    }
+
+    // A table of NetworkVariables that constitutes a Snapshot.
```

Comment:
"That constitues a Snapshot"
Why not name it "Snapshot" instead of EntryBlock?

Reply:
Good point. Will do